client/pages/search-form.jsx

```diff
@@ -0,0 +1,159 @@
+import React from 'react';
+
+export default class SearchForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      keywords: '',
+      language: '',
+      country: '',
+      articles: null
+    };
+    this.handleChangeKeyword = this.handleChangeKeyword.bind(this);
+    this.handleChangeLanguage = this.handleChangeLanguage.bind(this);
+    this.handleChangeCountry = this.handleChangeCountry.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChangeLanguage(event) {
+    this.setState({ language: event.target.value });
+  }
+
+  handleChangeCountry(event) {
+    this.setState({ country: event.target.value });
+  }
+
+  handleChangeKeyword(event) {
+    this.setState({ keywords: event.target.value });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    let lang = '&lang=';
+    let country = '&country=';
+    let keyString = '';
+    const keyArray = this.state.keywords.split(' ');
+    keyArray.map(keyword => {
+      if (keyword === keyArray[0]) {
+        keyString += `q=${keyword}`;
+        return keyString;
+      } else {
+        keyString += `&q=${keyword}`;
+        return keyString;
+      }
+    });
+    if (this.state.language !== '') {
+      lang += this.state.language;
+    } else {
+      lang = '';
+    }
+    if (this.state.country !== '') {
+      country += this.state.country;
+    } else {
+      country = '';
+    }
```

Comment:
Ok, I believe we can simplify this a lot by using a `URLSearchParams` object. Also, in general, don't use `map()` just for looping. That's what `forEach()` is for. Use `map()` when you want the returned array.

Anyway, you can conditionally add things to your URLSearchParams like this

```js
const searchParams = new URLSearchParams();
if (this.state.language !== '') {
  searchParams.set('lang', this.state.language);
}
if (this.state.country !== '') {
  searchParams.set('country', this.state.country);
}
// loop through the split key words and for each of them do
searchParams.append('q', aWord);

const queryString = searchParams.toString();
```